### PR TITLE
Dock widgets rebased

### PR DESCRIPTION
**Pull request by ljjh20**
Widgets live in docks. 
Main_widget skipped and live/timelapse instantiated directly in main_window. 
State-toggling slot refactored into separate functions for live/timelapse. 
Behaviour streamlined with split docks, TopDockAllowedArea, and placeholder central widget + resizing.

### CODE REVIEW

**Comment by ljjh20** [comment]
Let me know if this is closer to the behaviour we want.

File: widgets/gui/qt_main_window.py [hunk 1]
```diff
@@ -11,15 +15,59 @@ def __init__(self, *args, **kwargs):
...
+        self.timelapse_widget = TimelapseControl(self, "Timelapse Mode")
+        self.timelapse_dock.setWidget(self.timelapse_widget)
+        self.addDockWidget(Qt.RightDockWidgetArea, self.timelapse_dock)
+
+        self.parameters_widget = ParametersWidget(self)
+        self.parameters_dock.setWidget(self.parameters_widget)
+        self.addDockWidget(Qt.LeftDockWidgetArea, self.parameters_dock)
+
+        # split horizontal and vertical space between docks
+        self.splitDockWidget(self.parameters_dock, self.live_dock, Qt.Horizontal)
```
Comment (ljjh20) [inline widgets/gui/qt_main_window.py:57]: Also, let me know if this is a good way to split. I wasn't sure how to split it given live/timelapse take up the same horizontal space initially.

File: widgets/gui/qt_main_window.py [hunk 2]
```diff
@@ -11,15 +15,59 @@ def __init__(self, *args, **kwargs):
...
+        self.splitDockWidget(self.live_dock, self.timelapse_dock, Qt.Vertical)
```
Comment (ljjh20) [inline widgets/gui/qt_main_window.py:58]: Also, let me know if this is a good way to split. I wasn't sure how to split it given live/timelapse take up the same horizontal space initially.

**Comment by ljjh20** [comment]
I got it set up on Windows and gave it a test run. It ends up a lot smaller, so I resized things to stop squishing. The whole thing is still pretty small though so it may be worth adding the ability to resize isotropically.

**Review by AhmetCanSolak — COMMENTED** [review]
Looks good overall, i run on linux, my only concern is how to put dockable widgets back into window, i had hard time with parameters widget. Maybe we can put a little button on menubar to pull all dockable widgets back to their original positions? what do you think @ljjh20 ?

**Comment by ljjh20** [comment]
> Looks good overall, i run on linux, my only concern is how to put dockable widgets back into window, i had hard time with parameters widget. Maybe we can put a little button on menubar to pull all dockable widgets back to their original positions? what do you think @ljjh20 ?

I am pretty sure you can do that with the little Icon next to the X on each dock. Or is that not what you are trying to do?

**Comment by ljjh20** [comment]
Also, i find it works better with a slightly larger window / full screen. It attaches more willingly.

**Comment by AhmetCanSolak** [comment]
nope, X just terminates the widget on my linux machine, is that the behavior you get on OSX @ljjh20 ?

**Comment by ljjh20** [comment]
> nope, X just terminates the widget on my linux machine, is that the behavior you get on OSX @ljjh20 ?

What about the little icon to the left of the X?

**Comment by AhmetCanSolak** [comment]
icon next to it was not rendering due to other packages my environment, i changed the environment, now i saw the icon you are talking about. It seems working alright

**Comment by ljjh20** [comment]
> icon next to it was not rendering due to other packages my environment, i changed the environment, now i saw the icon you are talking about. It seems working alright

I can remove closable, I guess its not really needed.

File: widgets/gui/qt_main_window.py (hunk 2 above)
Comment (AhmetCanSolak) [reply widgets/gui/qt_main_window.py:54]: this way is good enough for now, has couple drawbacks but nothing major, just corner cases like when you try to dock back timelapse before live dock and so. We can keep it as is for now.

**Review by AhmetCanSolak — APPROVED** [review]
i think we are good, if you do not have anything else to add @ljjh20 we can merge this in.

**Comment by ljjh20** [comment]
> i think we are good, if you do not have anything else to add @ljjh20 we can merge this in.

Sure

**Pull request merged**